src/main/java/core/basesyntax/service/strategy/strategyimpl/ReturnOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.service.strategy.strategyimpl;
+
+import core.basesyntax.dao.Storage;
+import core.basesyntax.model.FruitRecord;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void apply(FruitRecord transaction) {
+        int returnFruits = transaction.getQuantity();
+        Storage.storage.merge(transaction.getFruit(),returnFruits, Integer::sum);
+
+        if (returnFruits < 0) {
+            throw new RuntimeException("Balance cannot be negative for fruit: "
+                    + transaction.getFruit());
+        }
```

Comment:
The check for a negative quantity should be done before applying the return operation. As it stands, the negative quantity would be merged into the storage before the exception is thrown, which could lead to incorrect storage state.